assets/src/edit-story/components/header/buttons.js

```diff
@@ -55,19 +59,94 @@ function PreviewButton() {
       meta: { isSaving },
       story: { link },
     },
+    actions: { saveStory },
   } = useStory();
 
+  const [previewLinkToOpenViaDialog, setPreviewLinkToOpenViaDialog] = useState(
+    null
+  );
+
   /**
    * Open a preview of the story in current window.
    */
   const openPreviewLink = () => {
     const previewLink = addQueryArgs(link, { preview: 'true' });
-    window.open(previewLink, 'story-preview');
+
+    // Start a about:blank popup with waiting message until we complete
+    // the saving operation. That way we will not bust the popup timeout.
+    let popup;
+    try {
+      popup = window.open('about:blank', PREVIEW_TARGET);
```

Comment:
Should we keep this window as a ref?

---

This might be a pre-existing issue, but what if someone goes to the preview page and navigates away? Nothing happens in that case right now 😕

Reply:
I don't think we need to keep the window as a ref. It's well localized to the event callback here.

With cross-domain navigation, there's not much we can do. One approach we could take is change the target ("story-preview" + random) in the `openPreviewLinkSync`. It's meant as a catch-all case so it's probably should be as safe as we can make it.

Reply:
Did that. PTAL.